torch/csrc/jit/tensorexpr/ir_simplifier.cpp

```diff
@@ -1345,6 +1345,29 @@ Stmt* IRSimplifierBase::mutate(const Cond* v) {
   return new Cond(cond_new, true_new, false_new);
 }
 
+Stmt* handleForCondReordering(const For* loop, Cond* cond) {
+  if (cond->false_stmt()) {
+    return nullptr;
+  }
+
+  auto condition_vars = VarFinder::find(cond->condition());
+  for (auto* v : condition_vars) {
+    // If the condition depends on the loop var, we cannot take it out of scope.
+    if (v == loop->var()) {
+      return nullptr;
+    }
+
+    // If the condition depends on a Var that is modified in the loop body, it
+    // may not be safe to reorder.
+    if (ModifiesVarChecker::check(cond, v)) {
```

Comment:
If you do `ModifiesVarChecker::check(loop, v)` instead, that should take care of checking if it is a loop variable as well, right? Then you can remove the above condition.

Reply:
I feel like there was a reason this didn't work but I can't think of it, let me experiment with it a bit.

Reply:
OK it didn't work because the ModifiesVarChecker didn't recurse into the body of a For. Easy fix.

Reply:
Great.